lib/evaluate.js

```diff
@@ -257,8 +257,9 @@ export const runFraudDetection = async (roundIndex, measurements, sparkRoundDeta
     assert(typeof m.inet_group === 'string', 'missing inet_group')
     assert(typeof m.finished_at === 'number', 'missing finished_at')
 
-    // TODO: check also Filecoin Deal id
-    const isValidTask = sparkRoundDetails.retrievalTasks.some(t => t.cid === m.cid)
+    const isValidTask = sparkRoundDetails.retrievalTasks.some(
+      t => t.cid === m.cid && t.minerId === m.minerId
```

Comment:
FYI: I first updated a test to fail with an expected error and only then made this change.